ktor-client/ktor-client-features/ktor-client-auth-basic/src/io/ktor/client/features/auth/basic/BasicAuth.kt

```diff
@@ -0,0 +1,47 @@
+package io.ktor.client.features.auth.basic
+
+import io.ktor.client.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.util.*
+
+class BasicAuth(val username: String, val password: String) {
+
+  class Configuration {
+
+    var username: String = ""
+    var password: String = ""
+
+    fun build(): BasicAuth = BasicAuth(username, password)
+  }
+
+  companion object Feature: HttpClientFeature<Configuration, BasicAuth> {
+
+    override val key: AttributeKey<BasicAuth> = AttributeKey("AuthBasicHeader")
+
+    override fun prepare(configure: Configuration.() -> Unit): BasicAuth = Configuration().apply(configure).build()
+
+    override fun install(feature: BasicAuth, scope: HttpClient) {
+
+      scope.requestPipeline.intercept(HttpRequestPipeline.Send) { request ->
```

Comment:
`Send` is used to send a request to a remote server. I'd prefer to use `State` phase for this purpose